File: gto-support-db/src/test/java/org/ccci/gto/android/common/db/BaseContractTest.java
package org.ccci.gto.android.common.db;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class BaseContractTest {
    @Test(expected = IllegalArgumentException.class)
    public void verifyUniqueIndexNoFields() throws Exception {
        BaseContract.uniqueIndex();
    }

    @Test
    public void verifyUniqueIndex() throws Exception {
        assertThat(BaseContract.uniqueIndex("field1").trim(), is("UNIQUE(field1)"));
        assertThat(BaseContract.uniqueIndex("field1", "field2").trim(), is("UNIQUE(field1,field2)"));
    }
}

File: gto-support-db/src/test/java/org/ccci/gto/android/common/db/BaseContractTest.kt
package org.ccci.gto.android.common.db

import org.ccci.gto.android.common.db.BaseContract.Companion.uniqueIndex
import org.junit.Assert.assertEquals
import org.junit.Test

class BaseContractTest {
    @Test(expected = IllegalArgumentException::class)
    fun verifyUniqueIndexNoFieldsThrowsException() {
        uniqueIndex()
    }

    @Test
    fun verifyUniqueIndex() {
        assertEquals("UNIQUE(field1)", uniqueIndex("field1"))
        assertEquals("UNIQUE(field1,field2)", uniqueIndex("field1", "field2"))
    }
}
